prow/cluster/hook_deployment.yaml

```diff
@@ -30,10 +30,10 @@ spec:
       labels:
         app: hook
     spec:
-      terminationGracePeriodSeconds: 30
+      terminationGracePeriodSeconds: 180
```

Comment:
> // Ignore SIGTERM so that we don't drop hooks when the pod is removed.

When kubelet sends SIGTERM to a pod, the pod is marked with a deletionTimestamp and removed from the load balancer, thus it does not receive new events. It should be easy to wait until all existing events are processed and then shutdown. We should also make sure that hook is always scaled up first on an update (ideally, d.spec.strategy.rollingParams.maxUnavailable is 0).

Opened https://github.com/kubernetes/test-infra/issues/3396